linker/shared/src/main/scala/org/scalajs/linker/Output.scala

```diff
@@ -0,0 +1,63 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js linker            **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2018, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.linker
+
+import org.scalajs.io.WritableVirtualBinaryFile
+
+import java.net.URI
+
+/** Output specification for a linker run.
+ *
+ *  @param jsFile The JavaScript file a [[Linker]] writes to.
+ *
+ *  @param sourceMap The sourceMap file the linker writes to. A [[Linker]] may
+ *      ignore this file. N.b. the [[StandardLinker]] will ignore it if
+ *      [[StandardLinker.Config.sourceMap]] is false. Further, a [[Linker]] must
+ *      not fail if this is not set, but rather not write a source map (even if
+ *      it is configured to write a source map).
+ *
+ *  @param sourceMapURI URI to reach the source map from the JavaScript file.
+ *      This is typically a relative URI but is not required. A [[Linker]]
+ *      should use this (to emit into [[jsFile]]), even if [[sourceMap]] is not
+ *      set.
+ *
+ *  @param jsFileURI URI to reach the JavaScript file from the source map. This
+ *      is typically a relative URI but is not required. A [[Linker]] may use
+ *      this even if [[sourceMap]] is not set, but it is typically meaningless.
+ */
```

Comment:
I'm not 100% convinced that this is a reasonable API, but it's the best I could come up with.